hal/src/rtl872x/ble_hal.cpp

```diff
@@ -2456,8 +2492,16 @@ void BleGap::handleConnectionStateChanged(uint8_t connHandle, T_GAP_CONN_STATE n
             evt.params.connected.info = &connection.info;
             notifyLinkEvent(evt);
             if (connecting_ && connection.info.role == BLE_ROLE_CENTRAL) {
-                connecting_ = false;
+                // Postpone setting connecting_ = false until MTU exchange happens (or times out)
+                // as by the standard you are not allowed to perform any reads/writes etc during an MTU exchange
+                SPARK_ASSERT(!os_timer_is_active(attMtuExchangeTimer_, nullptr));
+                os_timer_set_id(attMtuExchangeTimer_, (void*)(uintptr_t)connHandle);
```

Comment:
Is this a FreeRTOS feature to store data with a timer? I have not noticed this before

Reply:
Yup.